### ISSUE TITLE
miner update

### ISSUE THREAD

**Issue by ssska12** [UnamSanctam/SilentCryptoMiner#704]
Can you make the miner stop working when opening full screen applications?

**Comment by UnamSanctam** [comment]
Well it depends on what you mean by fullscreen, there are multiple different ways to detect if something is in fullscreen, here are the two most common that I know of:
1) Get the currently active window and check if the window has the same size as the screen.
2) Call `SHQueryUserNotificationState` and see if it's currently set to something like `QUNS_BUSY` (`QUNS_RUNNING_D3D_FULL_SCREEN`  isn't really used anymore)

But they also have their own problems, mainly:
1) This would trigger for any window that is fullscreen, including your browser or any other program.
2) This asks Windows through the API whether it's currently recommended to send notifications to the user (we would check if `QUNS_BUSY`), meaning things like the Windows notifications that you might be familiar with or any other similar thing. It's quite nice since most games triggers it, but the biggest problem is that this isn't always very "accurate" for our purposes, for example: maximizing Steam, opening any video to fullscreen and a few other things also triggers `QUNS_BUSY`. There used to be `QUNS_RUNNING_D3D_FULL_SCREEN ` but it's not very used anymore (couldn't trigger it at all in Windows 10). Adding this will also officially end Windows XP support since that function is only supported on Windows Vista and above.

So from these we can see that 1) wouldn't be very good, but 2) could work as long as you'd be willing to compromise with it also pausing the miner while things like any video (YouTube, Twitch, Netflix, any other) or when programs like Steam are in fullscreen. And that it would most likely officially end support for Windows XP (some have reported that the miners xmrig currently doesn't work for them in Windows XP but adding this would officially end it).

**Comment by ssska12** [comment]
I don’t think that people are constantly sitting on YouTube, so you can sacrifice this for the sake of convenience
